deepmd/pt/utils/dataloader.py

```diff
@@ -50,6 +51,7 @@ def setup_seed(seed) -> None:
     torch.manual_seed(seed)
     torch.cuda.manual_seed_all(seed)
     torch.backends.cudnn.deterministic = True
+    dp_random.seed(seed)
```

Comment:
should also seed with rank, i.e. [rank, seed]; see the example at https://numpy.org/doc/stable/reference/random/parallel.html#sequence-of-integer-seeds.

Reply:
Maybe in another PR fixing #4440?

Reply:
Currently the seed has been set to _seed from config file_ + rank, which is OK for the basic need.

Reply:
> Currently the seed has been set to seed from config file + rank

I don't get this point. It's done by TF but not PT.

Reply:
> Currently the seed has been set to _seed from config file_ + rank, which is OK for the basic need.

@njzjz here
https://github.com/deepmodeling/deepmd-kit/blob/e695a91ca6f7a1c9c830ab1c58b7b7a05db3da23/deepmd/pt/entrypoints/main.py#L141